tree/ntuple/v7/src/RPageStorageFile.cxx

```diff
@@ -140,6 +141,51 @@ ROOT::Experimental::Detail::RPageSinkFile::CommitSealedPageImpl(DescriptorId_t p
    return WriteSealedPage(sealedPage, bytesPacked);
 }
 
+std::vector<ROOT::Experimental::RNTupleLocator>
+ROOT::Experimental::Detail::RPageSinkFile::CommitSealedPageVImpl(std::span<RPageStorage::RSealedPageGroup> ranges)
+{
+   size_t size = 0, bytesPacked = 0;
+   for (auto &range : ranges) {
+      if (range.fFirst == range.fLast) {
+         // Skip empty ranges, they might not have a physical column ID!
+         continue;
+      }
+
+      const auto bitsOnStorage = RColumnElementBase::GetBitsOnStorage(
+         fDescriptorBuilder.GetDescriptor().GetColumnDescriptor(range.fPhysicalColumnId).GetModel().GetType());
+      for (auto sealedPageIt = range.fFirst; sealedPageIt != range.fLast; ++sealedPageIt) {
+         size += sealedPageIt->fSize;
+         bytesPacked += (bitsOnStorage * sealedPageIt->fNElements + 7) / 8;
+      }
+   }
+   if (size >= std::numeric_limits<std::int32_t>::max() || bytesPacked >= std::numeric_limits<std::int32_t>::max()) {
+      // Cannot fit it into one key, fall back to one key per page.
+      return RPagePersistentSink::CommitSealedPageVImpl(ranges);
+   }
```

Comment:
We should remember to remove this once we have large key support. Perhaps add a TODO.

Reply:
Done. Alternatively, we could also split the cluster into a smaller number of keys, but I thought this is too much complication to implement...